### PR TITLE
TP-634 Architectural documentation.

### PR DESCRIPTION
**Pull request by simonwo**
An architecture primer. Best viewed in the rendered documentation, so would recommend checking out this branch and building it.

### CODE REVIEW

**Review by NDevox — COMMENTED** [review]
Overall looks pretty good. Just a couple of comments.

File: certificates/__init__.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+"""
+Certificates, licenses, and non-paper conditions are sometimes required to bring
+goods through customs.
+
+A certificate is not a representation of the actual document itself – our
```
Comment (andyhd) [inline certificates/__init__.py:5]: Did we say we were avoiding "we" and "our"? Should this be "the database stores"?

**Review by andyhd — APPROVED** [review]
This is so good 👍

**Comment by stuaxo** [comment]
It looks good, definitely a fan of the way this partially comes from the docstrings too.

**Pull request merged**